src/bson_compat.rs

```diff
@@ -29,20 +29,18 @@ pub(crate) fn cstr_to_str(cs: &CStr) -> &str {
     }
 }
 
+#[cfg(feature = "bson-3")]
 pub(crate) trait RawDocumentBufExt: Sized {
-    fn append_ref_compat<'a>(
+    fn append_ref<'a>(
```

Comment:
This was unrelated but I realized I could drop the ugly `_compat` suffix by having this trait only be present when using `bson` 3.0.